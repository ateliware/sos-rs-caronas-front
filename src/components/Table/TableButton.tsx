import { PropsWithChildren } from 'react';

type Props = {
  active?: boolean;
  onClick?: () => void;
};

export default function TableButton(props: PropsWithChildren<Props>) {
  const buttonClasses = [
    'btn',
    'btn--icon-square',
    'btn--table-action',
    'ml-s-100',
  ];
  if (props.active) buttonClasses.push('btn--table-action--active');

  return (
    <button onClick={props.onClick} className={buttonClasses.join(' ')}>
      <span className="btn__body">
        <span className="btn__body__icon material-symbols-rounded">
          {props.children}
        </span>
      </span>
    </button>
  );
}
