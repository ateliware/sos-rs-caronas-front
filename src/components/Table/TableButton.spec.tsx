import { fireEvent, render } from '@testing-library/react';

import TableButton from './TableButton';

describe('TableButton', () => {
  it('should render', () => {
    const { container } = render(<TableButton />);
    expect(container).toMatchSnapshot();
  });

  it('should render with variants', () => {
    const action = jest.fn();
    const { container } = render(
      <TableButton onClick={action}>add</TableButton>
    );
    expect(container).toMatchSnapshot();

    const button = container.querySelector('button') as Element;
    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(action).toHaveBeenCalled();
  });
});
