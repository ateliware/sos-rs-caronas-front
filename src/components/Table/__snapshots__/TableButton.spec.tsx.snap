// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableButton should render 1`] = `
<div>
  <button
    class="btn btn--icon-square btn--table-action ml-s-100"
  >
    <span
      class="btn__body"
    >
      <span
        class="btn__body__icon material-symbols-rounded"
      />
    </span>
  </button>
</div>
`;

exports[`TableButton should render with variants 1`] = `
<div>
  <button
    class="btn btn--icon-square btn--table-action ml-s-100"
  >
    <span
      class="btn__body"
    >
      <span
        class="btn__body__icon material-symbols-rounded"
      >
        add
      </span>
    </span>
  </button>
</div>
`;
